src/core/src/core_logic/PatchInstaller.py

```diff
@@ -260,6 +343,138 @@ def install_updates(self, maintenance_window, package_manager, simulate=False):
             self.status_handler.add_error_to_status(message, Constants.PatchOperationErrorCodes.OPERATION_FAILED)
             self.composite_logger.log_error(message)
 
+        return installed_update_count
+
+    def install_patches_in_batches(self, all_packages, packages, package_versions, maintenance_window, package_manager, simulate=False):
+        number_of_batches = int(math.ceil(len(packages) / float(Constants.PATCHING_BATCH_SIZE)))
```

Comment:
Example:
Packages: P1 P2 P3 P4 P5 P6 P7 
Batch Size: 3

Approach1:
batch1: P1 P2 P3 (P4 also got installed because P1 dependent on P4)
batch2: P4 P5 P6 (remove P4 as already installed)
new batch2: P5 P6 (only 2 packages instead of 3)
batch3: P7 

Approach2:
batch1: P1 P2 P3 (P4 also installed)
batch2: P5 P6 P7

Approach1: 3 batches
Approach2: 2 batches

As we discussed offline, we should go with less complicated approach in first iteration.